File: hard/20_longest_balanced_substring/longest_balanced_substring.test.js
const longestBalancedSubstring = require('./longest_balanced_substring.js');

describe('TestProgram', () => {
  it('Test Case 1', () => {
    const string = '(()))(';
    const expected = 4;
    const actual = longestBalancedSubstring(string);
    expect(actual).toEqual(expected);
  });
});

File: hard/20_longest_balanced_substring/longest_balanced_substring_test.py
import unittest
from longest_balanced_substring import LongestBalancedSubstring

class TestProgram(unittest.TestCase):
    def test_case_1(self):
        string = "(()))("
        expected = 4
        actual = LongestBalancedSubstring(string)
        self.assertEqual(actual, expected)
